README.md

```diff
@@ -116,18 +116,120 @@ One of the big concerns with IPFS, and even cloud data storage in general is enc
+
+Should you still want to do this, download the [install_from_source.sh script](./setup/scripts/misc/install_from_source.sh). This will ensure you have the proper go version, download the github repository, compile the cli, and install it to `/bin/temporal`.
+
+```bash
+#! /bin/bash
+
+# setup install variabels
+GOVERSION=$(go version | awk '{print $3}' | tr -d "go" | awk -F "." '{print $2}')
+WORKDIR="/tmp/temporal-workdir"
+# handle golagn version detection
+if [[ "$GOVERSION" -lt 11 ]]; then
+    echo "[ERROR] golang is less than 1.11 and will produce errors"
+    exit 1
+fi
+if [[ "$GOVERSION" -lt 12 ]]; then
+    echo "[WARN] detected golang version is less than 1.12 and may produce errors"
+fi
+# create working directory
+mkdir "$WORKDIR"
+cd "$WORKDIR"
+# download temporal
+git clone https://github.com/RTradeLtd/Temporal.git
+cd Temporal
+# initialize submodules, and download all dependencies
+make setup
+# make cli binary
+make install
+```
+
+**Downloading Pre-Built Binaries:**
+
+To download pre-built binaries, currently available for Linux and Mac OSX platforms, head on over to our [github releases page](https://github.com/RTradeLtd/Temporal/releases/latest).
+
+Download a binary for your appropriate platform, additionally you can download sha256 checksums of the released binaries for post-download integrity verification. 
+
+You'll then want to copy the pre-built binary over to anywhere in your `PATH` environment variable.
+
+Example for downloading and installing the Linux version:
+
+```bash
+# download the binary
+wget https://github.com/RTradeLtd/Temporal/releases/download/v2.2.7/temporal-v2.2.7-linux-amd64
+# download the binary checksum
+wget https://github.com/RTradeLtd/Temporal/releases/download/v2.2.7/temporal-v2.2.7-linux-amd64.sha256
+# store downloaded checksum output
+CK_HASH=$(cat *.sha256  | awk '{print $1}')
+# calculate sha256 checksum of downloaded binary
+DL_HASH=$(sha256sum temporal-v2.2.7-linux-amd64 | awk '{print $1}')
+# compare checksums, and copy file to `PATH` if ok
+# if this doesn't show ok then your binary is corrupted or has been tampered with
+if [[ "$CK_HASH" == "$DL_HASH" ]]; then  echo "ok" && sudo cp temporal-v2.2.7-linux-amd64 /usr/local/bin; fi
+```
+
+### Configuration Initialization
+
+After downloading Temporal, regardless of your setup process you will need a configuration file. To generate a config file at `/home/doggo/config.json` run the following command:
+```
+temporal -config /home/doggo/config.json init
+```
+**Alternatively you can set the environment variable `CONFIG_DAG` and `temporal init`, along with *all other commands* will read from this location. It is recommended that you do this as it makes using the cli a lot easier**
+
+It is **extremely** important you keep this in a directory that is only accessible to the users required to run the servivce as it contains usernames and passwords to key pieces of Temporal's infrastructure.
```

Comment:
this should be an open issue to fix IMHO, there are methods we can explore to obscure this. Such as key vaults ([here is a azure example](https://docs.microsoft.com/en-us/rest/api/keyvault/))

Reply:
It definitely should be, I will create a github issue about it.

Reply:
Also we could require the files to have only a limited ser of permissions, like some sensitive files used by other services